woocommerce/woocommerce-android
# [Scan to Update Inventory] Add undo option to success snackbar

Closes: #10285 


### Description


This PR replaces the original Success snackbar on quantity increased with an Undo Snackbar that reverts the increase. It also adds a second snackbar after the undo option is clicked to cue the user that the action has been reverted. 

** Please note a bunch of tests have been changed but just to fix detekt issues.

### Testing instructions


- Open product list
- Click "barcode scanner" icon
- Scan barcode with the SKU of existing stock-managed product (you can use online barcode generator like [this one](https://barcode.tec-it.com/en/?data=ACGADO))
- Verify product is found and the bottom sheet is opened
- change the quantity and verify the product quantity is updated successfully
- Verify the Undo snackbar is displayed
- click on the undo button
- verify the second snackbar is displayed telling the user the action has been reverted
- scan the code again to verify the quantity is the same as it was before the increase. 

### Images/gif



https://github.com/woocommerce/woocommerce-android/assets/30724184/649cfe5a-639f-4a94-b14d-7da7ed4744f8
